src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -266,6 +269,18 @@ public ListBoxModel doFillTlsConfigurationItems() {
             return model;
         }
 
+        /**
+         * Displays an error message if the provided password is less than 14 characters
+         * while in FIPS mode. This message is triggered when the bindPassword field loses focus.
+         */
+        @RequirePOST
+        public FormValidation doCheckBindPassword(@QueryParameter String bindPassword) {
+            if(FIPS140.useCompliantAlgorithms() && StringUtils.length(bindPassword) < 14) {
+                return FormValidation.error(Messages.passwordTooShortFIPS());
+            }
+            return FormValidation.ok();
+        }
+
```

Comment:
Missing a `readResolve` check as we've done for the other fixes.

Reply:
as stated in several places, `readResolve` is not even needed as FIPS doesn't support backward compatibility to begin with

Reply:
There is already one such check for tlsCert in this plugin so we should try to be consistent.

Reply:
one check that I already explained it wasn't necessary and anyway you insisted to include making now needed that artificial consistency. Anyway, on my side I'm not converting this thread in a never ending discussion. The check was included yesterday and I'm resolving the conversation